src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -1272,6 +1272,21 @@ export class EditorPart extends Part implements IEditorPart, IEditorGroupService
 			return Position.ONE; // can only be ONE
 		}
 
+		const config = this.configurationService.getConfiguration<IWorkbenchEditorConfiguration>();
+		const reuseIfOpen = config.workbench.editor.reuseIfOpen;
+		// Respect option to reveal an editor if it is open (not necessarily already visible)
+		const skipReuse = (options && options.index) || arg1;
```

Comment:
There are imho more cases where you need to `skipReuse`: when `revealIfVisible` is true and there is a visible editor matching the input you should prefer it